Python/dp/Knapsack_fractional.py

```diff
@@ -0,0 +1,49 @@
+""" 
+Knapsack problem using the fractional/greedy method
+given the weights and their corresponding profit values, 
+fractional implementation allows filling knapsacks with parts/fractions of the items
+We fill knapsack of capacity W to obtain maximum possible value in bottom-up manner.
+Time Complexity = O(n logn)
+"""
+def knapsack_fractional(weights,values,capacity):
+    
+    #initialisaing the value of max_profit variable
+    max_profit=0
+    
+    for pair in sorted(zip(weights,values),key=lambda x:-x[1]/x[0]): # sorting the pair of values in descending order
+       
+        #if weight of highsest pair is greater than capacity, the amount is added in fractions
+        if pair[0]>capacity:
+           # while((pair[1]/(pair[0]/capacity))!=0)
+            max_profit+=int(pair[1]/(pair[0]/capacity))
+            capacity=0
+       
+        #if highest pair is lesser than capacity then the next pair is also added in fractions    
+        elif pair[0]<= capacity:
+            max_profit+=pair[1]
+            capacity-=pair[0]
+    
+    #returns nearest possible integer value of profit
+    return int(max_profit) 
+
```

Comment:
Format your Code

Reply:
I will add the rest changes. how should I format my code can you please explain more, I'm not able to understand